client/src/contexts/MealContext.js

```diff
@@ -0,0 +1,46 @@
+import React, { useReducer } from "react";
+import { useHistory, useLocation } from "react-router-dom";
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "createMeal":
+      return { meals: [...state.meals, action.payload.createdMeal], errorMessage: "" };
+    default:
+      return state;
+  }
+};
+
+const Context = React.createContext();
+
+const Provider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, {
+    meals: [],
+    errorMessage: "",
+  });
+
+  let history = useHistory();
+  let location = useLocation();
+
+
+  const createMeal = async({mealName, description}) => {
+    // do backend call
+    dispatch({ 
+      type: "createMeal", 
+      payload: { createdMeal: 
+        { mealName, description }
+    }});
+    // go back to previous page
+    let { from } = location.state || { from: { pathname: "/" } };
```

Comment:
why do we need to navigate back here ? we create a meal using a dialog right ? that does not affect the current route if am not mistaken

Reply:
true, i will change this.